video-morph/src/components/store/transcodeSlice.js

```diff
@@ -0,0 +1,76 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { createFFmpeg, fetchFile } from '@ffmpeg/ffmpeg';
+
+const ffmpeg = createFFmpeg({ log: true });
+
+export const loadFFmpeg = createAsyncThunk('transcode/loadFFmpeg', async () => {
+  await ffmpeg.load();
+  return true;
+});
+
+export const transcodeVideo = createAsyncThunk(
+  'transcode/transcodeVideo',
+  async ({ video, format }, { rejectWithValue }) => {
+    try {
+      ffmpeg.FS('writeFile', 'input', await fetchFile(video));
+      await ffmpeg.run('-i', 'input', `output.${format}`);
+      const data = ffmpeg.FS('readFile', `output.${format}`);
+
+      return URL.createObjectURL(new Blob([data.buffer], { type: `video/${format}` }));
+    } catch (error) {
+      return rejectWithValue(error.message);
+    }
+  }
+);
+
+const transcodeSlice = createSlice({
+  name: 'transcode',
+  initialState: {
+    ready: false,
```

Comment:
`Feedback`

I assume the transcoding will most likely occur on one page (i.e the index or home page). The redux implementation will basically just keep track of `ready`, `video`, `output`, `progress`, `error` and `loading` updates on this single page. Using redux here might then be an overkill.

React's `useState` and `useEffect` should be able to handle these updates. You could also explore using [React's context API](https://react.dev/learn/scaling-up-with-reducer-and-context).

Reply:
all right ma,

Reply:
@efalayi wanted to inform you that we deployed the project on a separate repo because I was encountering some issues during deployment ma, and this is the live site video-morph-v1.vercel.app

Reply:
Thanks for the update @abdussalam-mustapha 👍🏼 
There are conflicts in this PR and the latest PR: [https://github.com/video-morph/video-morph-frontend/pull/2](https://github.com/video-morph/video-morph-frontend/pull/2). I think you should close both PRs and create a new new one (to avoid merge conflicts).

Reply:
@efalayi I already closed them, however I was not able to create a new PR for the main-transcoding branch, because I didn't push any latest code there, so the new repo is the video-morph-v1.

Also, kindly check if the site video-morph-v1.vercel.app displays as it is in the screenshot ma, because it looks like it is not fully accessible by everybody

Reply:
I don't have access to this new repo: video-morph-v1. Please could you share the link.

The site on [https://video-morph-v1.vercel.app/](https://video-morph-v1.vercel.app/) currently displays the screenshot below.

<img width="1440" alt="Screenshot 2024-06-27 at 13 34 44" src="https://github.com/video-morph/video-morph-frontend/assets/28438068/e11eb839-8255-4bc2-8229-9bf985d9f892">


No texts displayed. I'm currently using Google Chrome to view the site. Please could you confirm if this is the same on your end?

Reply:
Same view using Mozilla and running the app locally. Could you share a screenshot of the app running locally?

Reply:
So as it stands, it doe'nt display like that again from my end and even the live site because I already started using tsparticles to complement for lack of designs
![efa](https://github.com/video-morph/video-morph-frontend/assets/103737038/4c1563af-32e0-495c-942b-55533cf17413)

Reply:
@efalayi so the reason some of the texts are not displaying for now from my end is due to z-index which I am currently on ma

Reply:
Okay. All good then. I guess this is good to go after you merge your changes cc @jidesakin

Reply:
okay, I just changed the issue you mentioned ma, I would push from my end now ma

Reply:
I think the issue is fixed now ma @efalayi, thank youuu so much

Reply:
Cool. Looks good on my end 👍🏼